spec/dummy/app/views/welcome/accordion_example.html.erb

```diff
@@ -1,6 +1,7 @@
 <h1>A test page in the dummy app</h1>
 <p>This is for testing the accordion component</p>
 <%= render "govuk_publishing_components/components/accordion", {
+  event_tracking: true,
```

Comment:
Why does the dummy app need to be updated?

Reply:
So the plan, at the moment, is to [extend the testing](https://github.com/alphagov/govuk_publishing_components/blob/main/spec/features/accordion_spec.rb) so the dummy app loads GA and can check this event fires as expected.  I'm aware other testing does this a different way by testing a [copy of the element instead of the element](https://github.com/alphagov/govuk_publishing_components/blob/main/spec/javascripts/components/checkboxes-spec.js).

However I feel like these aren't real tests as we're almost trying to keep copies of the HTML fixture in sync rather than testing what the user actually sees.  As the Accordion testing happens on an actual page, I thought it make more sense to continue with this approach.  (If I create a new `accordion-spec.js` file to test tracking i'm un-doing this work)

Reply:
Okay, that makes sense, but I'd save this change for that future work - it doesn't add anything to this PR.

Reply:
Sure, happy to capture to tackle outside of this PR

Reply:
https://github.com/alphagov/govuk_publishing_components/issues/2694

Reply:
Sorry, I think I misunderstood this slightly and I've looked at the code again. This component, despite having a JS file, does not appear to have a corresponding JS test file. That is not good, but it is not your fault. However, since you're adding some JS to this file, I think this is the right moment to create a JS test file and put some tests in for this code, although I don't expect you to complete the file by adding any further tests for the rest of the already existing JS.

I think also that the [issue you raised](https://github.com/alphagov/govuk_publishing_components/issues/2694) should be updated to more simply state that the whole of the JS for the accordion should be tested. In terms of the point raised about testing against the actual markup of the component vs a copy of the markup, that's valid, but we don't currently have a mechanism for that, and there is still value in testing the JS in this way. For example, the first thing the accordion JS does is add a class to the accordion - which can be tested fully regardless of whether it is run against the real markup of the component or a copy.

Reply:
> This component, despite having a JS file, does not appear to have a corresponding JS test file. That is not good, but it is not your fault. However, since you're adding some JS to this file, I think this is the right moment to create a JS test file and put some tests in for this code, although I don't expect you to complete the file by adding any further tests for the rest of the already existing JS.

> I think also that the https://github.com/alphagov/govuk_publishing_components/issues/2694 should be updated to more simply state that the whole of the JS for the accordion should be tested. 

[So, there's a lot of testing happening on the Design System side](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/accordion/accordion.test.js), I didn't wish to duplicate testing here as it's redundant. "gem-only" features that exist here have been layered on top and these are [tested](https://github.com/alphagov/govuk_publishing_components/blob/main/spec/features/accordion_spec.rb).  I'm wondering for this GA tracking to restore this file.

Is this a comment to add testing to re-test these via the JS?

> In terms of the point raised about testing against the actual markup of the component vs a copy of the markup, that's valid, but we don't currently have a mechanism for that, 

As this is now rendering on the `dummy_app` (http://govuk-publishing-components.dev.gov.uk/accordionexample) is this not a valid mechanism? (We use this for `/tabsexample` as well?)

> and there is still value in testing the JS in this way.  For example, the first thing the accordion JS does is add a class to the accordion - which can be tested fully regardless 

So, this has been tested here, against rendered markup,  Should I duplicate this in a JS file?
https://github.com/alphagov/govuk_publishing_components/blob/main/spec/features/accordion_spec.rb#L24

> of whether it is run against the real markup of the component or a copy.

In this context it's against the rendered accordion, I'm wondering what coverage is perhaps missing from the testing here:  https://github.com/alphagov/govuk_publishing_components/blob/main/spec/features/accordion_spec.rb

Reply:
I think I haven't made myself clear. You've added some JS functionality and that should have a corresponding test. Something like:

```
// accordion-spec.js

var FIXTURE = document.createElement('div')
new GOVUK.Modules.accordion(FIXTURE).init()
spyOn(window.GOVUK.analytics, 'trackEvent')

var showAllLink = FIXTURE.querySelector('.show-all-link')
showAllLink.click()
expect(window.GOVUK.analytics.trackEvent).toHaveBeenCalledWith('whatever') // show all

showAllLink.click()
expect(window.GOVUK.analytics.trackEvent).toHaveBeenCalledWith('whatever') // hide all

```

Reply:
Ace, I'm on this page - I'm restoring the file and doing something similar just for tracking